ipaserver/install/installutils.py

```diff
@@ -259,6 +291,32 @@ def record_in_hosts(ip, host_name=None, conf_file=paths.HOSTS):
 
     return None
 
+
+def order_hosts_file(host_name):
+    """Read and overwrite /etc/hosts with FQDN at beginning, issue #8018"""
+    normalize_host = ''.join(host_name)
+    hostsfile = '/etc/hosts'
+    newfile = '/tmp/hosts'
```

Comment:
Please use a temporary file instead, no predicted `newfile`.

Reply:
And create the temporary file on the same file system as the ``/etc/hosts`` so the rename can be atomic.

Reply:
done, using tempfile.NamedTemporaryFile() now